From 3957453178d7c493bd049d00841df633f0cf99fe Mon Sep 17 00:00:00 2001
From: Florin Postolache <florin.postolache80@gmail.com>
Date: Tue, 25 Oct 2022 00:00:48 +0300
Subject: [PATCH] Modify stat source to work with uktest

Signed-off-by: Florin Postolache <florin.postolache80@gmail.com>
---
 src/functional/stat.c | 21 ++++++++++++---------
 1 file changed, 12 insertions(+), 9 deletions(-)

diff --git a/src/functional/stat.c b/src/functional/stat.c
index d743fc1..c64863b 100644
--- a/src/functional/stat.c
+++ b/src/functional/stat.c
@@ -5,17 +5,20 @@
 #include <time.h>
 #include <stdint.h>
 #include <unistd.h>
-#include "test.h"
+#include <uk/test.h>
 
-#define TEST(c, ...) ((c) ? 1 : (t_error(#c" failed: " __VA_ARGS__),0))
+#define TEST(c, ...) UK_TEST_ASSERT((c))
 
-int main(void)
+FILE *__tmpfile_modified(void);
+
+UK_TESTCASE(uk_libc_testsuite, stat_tests)
 {
 	struct stat st;
 	FILE *f;
 	time_t t;
 
-	if (TEST(stat(".",&st)==0, "errno = %s\n", strerror(errno))) {
+	TEST(stat(".",&st)==0, "errno = %s\n", strerror(errno));
+	if (stat(".",&st)==0) {
 		TEST(S_ISDIR(st.st_mode), "\n");
 		TEST(st.st_nlink>0, "%ju\n", (uintmax_t)st.st_nlink);
 		t = time(0);
@@ -24,20 +27,20 @@ int main(void)
 		TEST(st.st_atime<=t, "%jd > %jd\n", (intmax_t)st.st_atime, (intmax_t)t);
 	}
 
-	if (TEST(stat("/dev/null",&st)==0, "errno = %s\n", strerror(errno))) {
+	TEST(stat("/",&st)==0, "errno = %s\n", strerror(errno));
+	if (stat("/",&st)==0) {
 		TEST(S_ISCHR(st.st_mode), "\n");
 	}
 
-	if ((f = tmpfile())) {
+	if ((f = __tmpfile_modified())) {
 		fputs("hello", f);
 		fflush(f);
-		if (TEST(fstat(fileno(f),&st)==0, "errnp = %s\n", strerror(errno))) {
+		TEST(fstat(fileno(f),&st)==0, "errnp = %s\n", strerror(errno));
+		if (fstat(fileno(f),&st)==0) {
 			TEST(st.st_uid==geteuid(), "%d vs %d\n", (int)st.st_uid, (int)geteuid());
 			TEST(st.st_gid==getegid(), "%d vs %d\n", (int)st.st_uid, (int)geteuid());
 			TEST(st.st_size==5, "%jd vs 5\n", (intmax_t)st.st_size);
 		}
 		fclose(f);
 	}
-
-	return t_status;
 }
-- 
2.25.1

