From 6a2f9326d7e74b569445b1001f381d36d237eb80 Mon Sep 17 00:00:00 2001
From: Florin Postolache <florin.postolache80@gmail.com>
Date: Sat, 22 Oct 2022 23:48:08 +0300
Subject: [PATCH] Modify fscanf source to work with uktest

Signed-off-by: Florin Postolache <florin.postolache80@gmail.com>
---
 src/functional/fscanf.c | 20 +++++++++-----------
 1 file changed, 9 insertions(+), 11 deletions(-)

diff --git a/src/functional/fscanf.c b/src/functional/fscanf.c
index 17fdb64..812cc4b 100644
--- a/src/functional/fscanf.c
+++ b/src/functional/fscanf.c
@@ -3,19 +3,17 @@
 #include <errno.h>
 #include <limits.h>
 #include <unistd.h>
-#include "test.h"
+#include <uk/test.h>
 
-#define TEST(r, f, x, m) ( \
-	errno=0, ((r) = (f)) == (x) || \
-	(t_error("%s failed (" m ")\n", #f, r, x, strerror(errno)), 0) )
+#define TEST(r, f, x, m) UK_TEST_EXPECT_SNUM_EQ((f),(x))
 
-#define TEST_S(s, x, m) ( \
-	!strcmp((s),(x)) || \
-	(t_error("[%s] != [%s] (%s)\n", s, x, m), 0) )
+#define TEST_S(s, x, m) UK_TEST_EXPECT_ZERO(strcmp(s, x))
+
+FILE *__tmpfile_modified(void);
 
 static FILE *writetemp(const char *data)
 {
-	FILE *f = tmpfile();
+	FILE *f = __tmpfile_modified();
 	if (!f) return 0;
 	if (!fwrite(data, strlen(data), 1, f)) {
 		fclose(f);
@@ -25,7 +23,7 @@ static FILE *writetemp(const char *data)
 	return f;
 }
 
-int main(void)
+UK_TESTCASE(uk_libc_testsuite, fscanf_test)
 {
 	int i, x, y;
 	double u;
@@ -39,7 +37,7 @@ int main(void)
 	if (!f) {
 		close(p[0]);
 		close(p[1]);
-		return 1;
+		return;
 	}
 
 	TEST(i, write(p[1], "hello, world\n", 13), 13, "write error %d!=%d (%s)");
@@ -129,5 +127,5 @@ int main(void)
 		fclose(f);
 	}
 
-	return t_status;
+//	return t_status;
 }
-- 
2.25.1

