From 4ad8acdc712a45aa163b9da61cc47e31596539f7 Mon Sep 17 00:00:00 2001
From: Florin Postolache <florin.postolache80@gmail.com>
Date: Mon, 14 Nov 2022 21:26:50 +0200
Subject: [PATCH] Modify lrand48-signextend source to work with uktest

Signed-off-by: Florin Postolache <florin.postolache80@gmail.com>
---
 src/regression/lrand48-signextend.c | 17 +++++++----------
 1 file changed, 7 insertions(+), 10 deletions(-)

diff --git a/src/regression/lrand48-signextend.c b/src/regression/lrand48-signextend.c
index dedd515..af0642a 100644
--- a/src/regression/lrand48-signextend.c
+++ b/src/regression/lrand48-signextend.c
@@ -1,16 +1,13 @@
 // lrand48 should give deterministic results
 #define _XOPEN_SOURCE 700
 #include <stdlib.h>
-#include "test.h"
+#include <uk/test.h>
 
-int main(void)
+UK_TESTCASE(uk_libc_random_tests, lrand48_signextend_tests)
 {
-	long r;
-	r = lrand48();
-	if (r != 0) t_error("1st lrand48() got %ld want 0\n", r);
-	r = lrand48();
-	if (r != 2116118) t_error("2nd lrand48() got %ld want 2116118\n", r);
-	r = lrand48();
-	if (r != 89401895) t_error("3rd lrand48() got %ld want 89401895\n", r);
-	return t_status;
+	UK_TEST_EXPECT_ZERO(lrand48());
+	
+	UK_TEST_EXPECT_SNUM_EQ(lrand48(), 2116118);
+	
+	UK_TEST_EXPECT_SNUM_EQ(lrand48(), 89401895);
 }
-- 
2.25.1

