From 4dabed0bd6fbf6e6bf7e77ae8abf0fc4fa5af8fb Mon Sep 17 00:00:00 2001
From: Florin Postolache <florin.postolache80@gmail.com>
Date: Thu, 10 Nov 2022 10:11:06 +0200
Subject: [PATCH] Modify strtof source to work with uktest

Signed-off-by: Florin Postolache <florin.postolache80@gmail.com>
---
 src/functional/strtof.c | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/src/functional/strtof.c b/src/functional/strtof.c
index a998bea..48cb18a 100644
--- a/src/functional/strtof.c
+++ b/src/functional/strtof.c
@@ -1,7 +1,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <math.h>
-#include "test.h"
+#include <uk/test.h>
 
 #define length(x) (sizeof(x) / sizeof *(x))
 
@@ -27,7 +27,7 @@ static struct {
 	{"340282356779733661637539395458142568448", INFINITY},
 };
 
-int main(void)
+UK_TESTCASE(uk_libc_conversion_tests, strtof_tests)
 {
 	int i;
 	float x;
@@ -35,8 +35,6 @@ int main(void)
 
 	for (i = 0; i < length(t); i++) {
 		x = strtof(t[i].s, &p);
-		if (x != t[i].f)
-			t_error("strtof(\"%s\") want %a got %a\n", t[i].s, t[i].f, x);
+		UK_TEST_ASSERTF((x == t[i].f), "strtof(%.7s, &p) == %.7f", t[i].s, x);
 	}
-	return t_status;
 }
-- 
2.25.1

