From 5001f18cea25a39c8650e9e6ff51b4e69381bff4 Mon Sep 17 00:00:00 2001
From: Florin Postolache <florin.postolache80@gmail.com>
Date: Fri, 28 Oct 2022 00:03:15 +0300
Subject: [PATCH] Modify mkdtemp-failure source to work with uktest

Signed-off-by: Florin Postolache <florin.postolache80@gmail.com>
---
 src/regression/mkdtemp-failure.c | 17 ++++++-----------
 1 file changed, 6 insertions(+), 11 deletions(-)

diff --git a/src/regression/mkdtemp-failure.c b/src/regression/mkdtemp-failure.c
index fa54c13..14ae061 100644
--- a/src/regression/mkdtemp-failure.c
+++ b/src/regression/mkdtemp-failure.c
@@ -7,24 +7,19 @@
 #include <stdlib.h>
 #include <string.h>
 #include <errno.h>
-#include "test.h"
+#include <uk/test.h>
 
 char *mkdtemp(char *);
 
-#define S "/dev/null/fooXXXX"
+#define S "/fooXXXX"
 
-int main(void)
+UK_TESTCASE(uk_libc_testsuite, mkdtemp_failure_tests)
 {
 	char p[] = S;
 	char *r;
 
 	r = mkdtemp(p);
-	if (r)
-		t_error("mkdtemp(" S ") did not fail\n");
-	if (memcmp(p, S, sizeof p) != 0)
-		t_error("mkdtemp(" S ") modified the template: %s\n", p);
-	if (r == 0 && errno != EINVAL)
-		t_error("mkdtemp(" S ") failed with %d [%s] instead of %d [%s]\n",
-			errno, strerror(errno), EINVAL, strerror(EINVAL));
-	return t_status;
+	UK_TEST_EXPECT_NULL(r);
+	UK_TEST_EXPECT_ZERO(memcmp(p, S, sizeof p));
+	UK_TEST_ASSERT(!(r == 0 && errno != EINVAL));
 }
-- 
2.25.1

