From 668ee5dc6acf88273da52857d4e5524c17b9a570 Mon Sep 17 00:00:00 2001
From: Florin Postolache <florin.postolache80@gmail.com>
Date: Thu, 10 Nov 2022 10:06:08 +0200
Subject: [PATCH] Modify strtod source to work with uktest

Signed-off-by: Florin Postolache <florin.postolache80@gmail.com>
---
 src/functional/strtod.c | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/src/functional/strtod.c b/src/functional/strtod.c
index 2e4b3fd..b3d99a9 100644
--- a/src/functional/strtod.c
+++ b/src/functional/strtod.c
@@ -1,7 +1,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <math.h>
-#include "test.h"
+#include <uk/test.h>
 
 #define length(x) (sizeof(x) / sizeof *(x))
 
@@ -59,7 +59,7 @@ static struct {
 	{"0.4996908522051874110779982354932499499602e9", 499690852.20518744}, // 0x1.dc8ad6434872ap+28
 };
 
-int main(void)
+UK_TESTCASE(uk_libc_conversion_tests, strtod_tests)
 {
 	int i;
 	double x;
@@ -67,9 +67,7 @@ int main(void)
 
 	for (i = 0; i < length(t); i++) {
 		x = strtod(t[i].s, &p);
-		if (x != t[i].f)
-			t_error("strtod(\"%s\") want %a got %a\n", t[i].s, t[i].f, x);
+		UK_TEST_ASSERTF((x == t[i].f), "strtod(%.7s, &p) == %.7lf", t[i].s, x);
 	}
-	return t_status;
 }
 
-- 
2.25.1

