From 9f696055dbadd2bc029d7abc892e811539bb7129 Mon Sep 17 00:00:00 2001
From: Florin Postolache <florin.postolache80@gmail.com>
Date: Mon, 14 Nov 2022 19:26:53 +0200
Subject: [PATCH] Modify crypt source to work with uktest

Signed-off-by: Florin Postolache <florin.postolache80@gmail.com>
---
 src/functional/crypt.c | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/src/functional/crypt.c b/src/functional/crypt.c
index 9629651..3da3fc3 100644
--- a/src/functional/crypt.c
+++ b/src/functional/crypt.c
@@ -1,14 +1,14 @@
 #include <string.h>
 #include <crypt.h>
-#include "test.h"
+#include <uk/test.h>
 
 static char *p;
 
 #define T(h,s,k) p = crypt(k,s); \
  if (!p) p ="*"; \
- if (strcmp(p,h)!=0) t_error("crypt(%s, \"%s\") failed: got \"%s\" want \"%s\"\n", #k, s, p, h);
+ UK_TEST_ASSERTF(strcmp(p,h) == 0, "Expected the coded message to be: %s", h);
 
-int main()
+UK_TESTCASE(uk_libc_crypt_tests, crypt_tests)
 {
 
 /* md5 */
@@ -104,6 +104,5 @@ T("$6$rounds=1000$roundstoolow$kUMsbe306n21p9R.FRkW3IGn.S9NPN0x50YhH1xhLsPuWGsUS
 //T("$6$rounds=123456$asaltof16chars..$BtCwjqMJGx5hrJhZywWvt0RLE8uZ4oPwcelCjmw2kSYu.Ec6ycULevoBK25fs2xXgMNrCzIMVcgEJAstJeonj1",
 //	"$6$rounds=123456$asaltof16chars..", "a short string")
 
-return t_status;
-}
 
+}
-- 
2.25.1

