From f3f0764df39c4ba71dee0f9fa8924f4f3dc96f51 Mon Sep 17 00:00:00 2001
From: Florin Postolache <florin.postolache80@gmail.com>
Date: Mon, 24 Oct 2022 23:21:10 +0300
Subject: [PATCH] Modify sscanf_long source to work with uktest

Signed-off-by: Florin Postolache <florin.postolache80@gmail.com>
---
 src/functional/sscanf_long.c | 17 ++++++-----------
 1 file changed, 6 insertions(+), 11 deletions(-)

diff --git a/src/functional/sscanf_long.c b/src/functional/sscanf_long.c
index c9208af..8e2a299 100644
--- a/src/functional/sscanf_long.c
+++ b/src/functional/sscanf_long.c
@@ -4,9 +4,9 @@
 #include <math.h>
 #include <errno.h>
 #include <sys/resource.h>
-#include "test.h"
+#include <uk/test.h>
 
-int main(void)
+UK_TESTCASE(uk_libc_testsuite, sscanf_long_tests)
 {
 	enum {n = 8*1024*1024};
 	char *s = malloc(n);
@@ -14,8 +14,7 @@ int main(void)
 	float f;
 	char c;
 
-	if (!s)
-		return t_error("out of memory");
+	UK_TEST_EXPECT_NOT_NULL(s);
 	t_setrlim(RLIMIT_STACK, 100*1024);
 
 	for (i = 0; i < n; i++) s[i] = '1';
@@ -29,12 +28,8 @@ int main(void)
 	 */
 	i = sscanf(s, "%f %c", &f, &c);
 
-	if (i != 2)
-		t_error("sscanf returned %d, want 2\n", i);
-	if (f != INFINITY)
-		t_error("sscanf(longnum, \"%%f\") read %f, want inf\n", f);
-	if (c != '1')
-		t_error("sscanf(\"1\", %%c) read '%c', want '1'\n", c);
+	UK_TEST_ASSERT((i == 2));
+	UK_TEST_ASSERT((f == INFINITY));
+	UK_TEST_ASSERT((c == '1'));
 	free(s);
-	return t_status;
 }
-- 
2.25.1

