From f8fe4eca73038cdc6033489521656ebbdc9e1500 Mon Sep 17 00:00:00 2001
From: Florin Postolache <florin.postolache80@gmail.com>
Date: Fri, 11 Nov 2022 10:16:46 +0200
Subject: [PATCH] Modify fdopen source to work with uktest

Signed-off-by: Florin Postolache <florin.postolache80@gmail.com>
---
 src/functional/fdopen.c | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/src/functional/fdopen.c b/src/functional/fdopen.c
index 88f7561..15ebbe0 100644
--- a/src/functional/fdopen.c
+++ b/src/functional/fdopen.c
@@ -3,33 +3,33 @@
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
-#include "test.h"
+#include <uk/test.h>
+#include "fs_essentials.h"
 
-#define TEST(c) do { \
-	errno = 0; \
-	if (!(c)) \
-		t_error("%s failed (errno = %d)\n", #c, errno); \
-} while(0)
+#define TEST(c) UK_TEST_ASSERT((c))
 
-int main(void)
+UK_TESTCASE(uk_libc_file_folder_tests, fdopen_tests)
 {
 	char tmp[] = "/tmp/testsuite-XXXXXX";
 	char foo[6];
 	int fd;
 	FILE *f;
 
+	UK_TEST_EXPECT_ZERO((fd = create_fs_for_tests()));
+	if(fd)
+		return;
+
 	TEST((fd = mkstemp(tmp)) > 2);
 	TEST(write(fd, "hello", 6)==6);
-	TEST(f = fdopen(fd, "rb"));
+	TEST((f = fdopen(fd, "rb")) != NULL);
 	if (f) {
 		TEST(ftello(f)==6);
 		TEST(fseeko(f, 0, SEEK_SET)==0);
 		TEST(fgets(foo, sizeof foo, f));
-		if (strcmp(foo,"hello") != 0)
-			t_error("fgets read back: \"%s\"; wanted: \"hello\"\n", foo);
+		TEST(!strcmp(foo,"hello") != 0);
 		fclose(f);
 	}
 	if (fd > 2)
 		TEST(unlink(tmp) != -1);
-	return t_status;
+	UK_TEST_EXPECT_ZERO(destroy_fs());
 }
-- 
2.25.1

