From e147ff6f9d7b9f11b75b69677e2699a5123bffcd Mon Sep 17 00:00:00 2001
From: Florin Postolache <florin.postolache80@gmail.com>
Date: Thu, 10 Nov 2022 10:29:07 +0200
Subject: [PATCH] Modify wcstol source to work with uktest

Signed-off-by: Florin Postolache <florin.postolache80@gmail.com>
---
 src/functional/wcstol.c | 20 ++++++++------------
 1 file changed, 8 insertions(+), 12 deletions(-)

diff --git a/src/functional/wcstol.c b/src/functional/wcstol.c
index 2f0e8a4..2017075 100644
--- a/src/functional/wcstol.c
+++ b/src/functional/wcstol.c
@@ -3,17 +3,16 @@
 #include <string.h>
 #include <errno.h>
 #include <wchar.h>
-#include "test.h"
+#include <uk/test.h>
 
-#define TEST(r, f, x, m) ( \
-	errno = 0, msg = #f, ((r) = (f)) == (x) || \
-	(t_error("%s failed (" m ")\n", #f, r, x), 0) )
+#define TEST(r, f, x, m) ({ \
+	errno = 0; \
+	UK_TEST_ASSERT(((f) == (x))); \
+})
 
-#define TEST2(r, f, x, m) ( \
-	((r) = (f)) == (x) || \
-	(t_error("%s failed (" m ")\n", msg, r, x), 0) )
+#define TEST2(r, f, x, m) UK_TEST_ASSERT(((f) == (x)))
 
-int main(void)
+UK_TESTCASE(uk_libc_conversion_tests, wcstol_tests)
 {
 	int i;
 	long l;
@@ -68,9 +67,7 @@ int main(void)
 		TEST(ul, wcstoul(s=L"-9223372036854775809", &c, 0), -9223372036854775809UL, "rejected negative %lu != %lu");
 		TEST2(i, c-s, 20, "wrong final position %d != %d");
 		TEST2(i, errno, 0, "spurious errno %d != %d");
-	} else {
-		t_error("sizeof(long) == %d, not implemented\n", (int)sizeof(long));
-	}
+	} 
 
 	TEST(l, wcstol(L"z", 0, 36), 35, "%ld != %ld");
 	TEST(l, wcstol(L"00010010001101000101011001111000", 0, 2), 0x12345678, "%ld != %ld");
@@ -85,5 +82,4 @@ int main(void)
 	TEST(l, wcstol(s=L"123", &c, 37), 0, "%ld != %ld");
 	TEST2(i, c-s, 0, "wrong final position %d != %d");
 	TEST2(i, errno, EINVAL, "%d != %d");
-	return t_status;
 }
-- 
2.25.1

