From 04cb85db43d873215799c1a2c60b2d8cc7130616 Mon Sep 17 00:00:00 2001
From: Florin Postolache <florin.postolache80@gmail.com>
Date: Mon, 14 Nov 2022 19:56:20 +0200
Subject: [PATCH] Modify dn_expand-empty source to work with uktest

Signed-off-by: Florin Postolache <florin.postolache80@gmail.com>
---
 src/regression/dn_expand-empty.c | 14 ++++----------
 1 file changed, 4 insertions(+), 10 deletions(-)

diff --git a/src/regression/dn_expand-empty.c b/src/regression/dn_expand-empty.c
index 2afe634..8600ab9 100644
--- a/src/regression/dn_expand-empty.c
+++ b/src/regression/dn_expand-empty.c
@@ -2,20 +2,14 @@
 #define _DEFAULT_SOURCE 1
 #define _BSD_SOURCE 1
 #include <resolv.h>
-#include "test.h"
+#include <uk/test.h>
 
-int main(void)
+UK_TESTCASE(uk_libc_network_tests, dn_expand_empty_tests)
 {
 	unsigned char packet[] = "";
 	char name[] = "XX";
-	int r;
 
 	/* empty name in packet */
-	r = dn_expand(packet, packet+1, packet, name, 1);
-	if (r!=1)
-		t_error("dn_expand(\"\", name, 1) returned %d, wanted 1\n", r);
-	if (name[0])
-		t_error("dn_expand(\"\", name, 1) failed: got \"%s\" name, wanted \"\"\n", name);
-
-	return t_status;
+	UK_TEST_ASSERT(dn_expand(packet, packet+1, packet, name, 1) == 1);
+	UK_TEST_ASSERT(name[0] == '\0');
 }
-- 
2.25.1

