From 0da44a674b11bf38487a37e69889928c6cfe483e Mon Sep 17 00:00:00 2001
From: Florin Postolache <florin.postolache80@gmail.com>
Date: Sun, 23 Oct 2022 00:20:42 +0300
Subject: [PATCH] Modify fcntl source to work with uktest

Signed-off-by: Florin Postolache <florin.postolache80@gmail.com>
---
 src/functional/fcntl.c | 19 +++++++++++--------
 1 file changed, 11 insertions(+), 8 deletions(-)

diff --git a/src/functional/fcntl.c b/src/functional/fcntl.c
index 485c00c..39c0342 100644
--- a/src/functional/fcntl.c
+++ b/src/functional/fcntl.c
@@ -4,12 +4,13 @@
 #include <errno.h>
 #include <string.h>
 #include <sys/wait.h>
-#include "test.h"
+#include <uk/test.h>
 
-#define TEST(c, ...) ((c) ? 1 : (t_error(#c" failed: " __VA_ARGS__),0))
-#define TESTE(c) (errno=0, TEST(c, "errno = %s\n", strerror(errno)))
+#define TEST(c, ...) UK_TEST_ASSERT(c)
 
-int main(void)
+FILE *__tmpfile_modified(void);
+
+UK_TESTCASE(uk_libc_testsuite, fcntl_tests)
 {
 	struct flock fl = {0};
 	FILE *f;
@@ -17,14 +18,18 @@ int main(void)
 	pid_t pid;
 	int status;
 
-	if (!TESTE(f=tmpfile())) return t_status;
+	errno = 0;
+	f=__tmpfile_modified();
+	UK_TEST_EXPECT_NOT_NULL(f);
+	if (!f) return;
 	fd = fileno(f);
 
 	fl.l_type = F_WRLCK;
 	fl.l_whence = SEEK_SET;
 	fl.l_start = 0;
 	fl.l_len = 0;
-	TESTE(fcntl(fd, F_SETLK, &fl)==0);
+	errno = 0;
+	UK_TEST_EXPECT_ZERO(fcntl(fd, F_SETLK, &fl));
 
 	pid = fork();
 	if (!pid) {
@@ -44,6 +49,4 @@ int main(void)
 	TEST(status==0, "child failed to detect lock held by parent\n");
 
 	fclose(f);
-
-	return t_status;
 }
-- 
2.25.1

