From 550461d03009c284d815daebecc6d80bb6220672 Mon Sep 17 00:00:00 2001
From: Florin Postolache <florin.postolache80@gmail.com>
Date: Sun, 7 Aug 2022 20:34:41 +0300
Subject: [PATCH] Modify fscanf source to work with uktest

Signed-off-by: Florin Postolache <florin.postolache80@gmail.com>
---
 src/functional/fscanf.c | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/src/functional/fscanf.c b/src/functional/fscanf.c
index 17fdb64..191006e 100644
--- a/src/functional/fscanf.c
+++ b/src/functional/fscanf.c
@@ -3,15 +3,15 @@
 #include <errno.h>
 #include <limits.h>
 #include <unistd.h>
-#include "test.h"
+#include <uk/test.h>//#include "test.h"
 
-#define TEST(r, f, x, m) ( \
-	errno=0, ((r) = (f)) == (x) || \
-	(t_error("%s failed (" m ")\n", #f, r, x, strerror(errno)), 0) )
+#define TEST(r, f, x, m) UK_TEST_EXPECT_SNUM_EQ((f),(x))//( \
+//	errno=0, ((r) = (f)) == (x) || \
+//	(t_error("%s failed (" m ")\n", #f, r, x, strerror(errno)), 0) )
 
-#define TEST_S(s, x, m) ( \
-	!strcmp((s),(x)) || \
-	(t_error("[%s] != [%s] (%s)\n", s, x, m), 0) )
+#define TEST_S(s, x, m) UK_TEST_EXPECT_ZERO(strcmp(s, x))//( \
+//	!strcmp((s),(x)) || \
+//	(t_error("[%s] != [%s] (%s)\n", s, x, m), 0) )
 
 static FILE *writetemp(const char *data)
 {
@@ -25,7 +25,7 @@ static FILE *writetemp(const char *data)
 	return f;
 }
 
-int main(void)
+UK_TESTCASE(uk_libc_testsuite, fscanf_test)//int main(void)
 {
 	int i, x, y;
 	double u;
@@ -39,7 +39,7 @@ int main(void)
 	if (!f) {
 		close(p[0]);
 		close(p[1]);
-		return 1;
+//		return 1;
 	}
 
 	TEST(i, write(p[1], "hello, world\n", 13), 13, "write error %d!=%d (%s)");
@@ -129,5 +129,5 @@ int main(void)
 		fclose(f);
 	}
 
-	return t_status;
+//	return t_status;
 }
-- 
2.25.1

