From 110ef691e704f5902025e6f2726de5df7e6babb1 Mon Sep 17 00:00:00 2001
From: Florin Postolache <florin.postolache80@gmail.com>
Date: Thu, 27 Oct 2022 10:06:06 +0300
Subject: [PATCH] Modify fgets-eof source to work with uktest

Signed-off-by: Florin Postolache <florin.postolache80@gmail.com>
---
 src/regression/fgets-eof.c | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/src/regression/fgets-eof.c b/src/regression/fgets-eof.c
index 4b3e41f..924d337 100644
--- a/src/regression/fgets-eof.c
+++ b/src/regression/fgets-eof.c
@@ -2,11 +2,11 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include "test.h"
+#include <uk/test.h>
 
-#define ASSERT(c) do { if (!(c)) t_error("%s failed\n", #c); } while(0)
+#define ASSERT(c) UK_TEST_ASSERT((c))
 
-int main(void)
+UK_TESTCASE(uk_libc_file_folder_tests, fgets_eof_tests)
 {
 	char buf[] = "test";
 	char s[10];
@@ -16,7 +16,5 @@ int main(void)
 	ASSERT(fgets(s, sizeof s, f) == s);
 	ASSERT(strcmp(s, buf) == 0);
 	ASSERT(fgets(s, sizeof s, f) == 0);
-	if (s[0] != 't')
-		t_error("fgets modified the buffer after eof\n");
-	return t_status;
+	UK_TEST_ASSERT(s[0] == 't');
 }
-- 
2.25.1

