From aa76fcf9ecfb292cf23aac266bcbaeb0939c8e2a Mon Sep 17 00:00:00 2001
From: Florin Postolache <florin.postolache80@gmail.com>
Date: Sun, 9 Oct 2022 22:21:13 +0300
Subject: [PATCH] Modify sem_open source to work with uktest

---
 src/functional/sem_open.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/functional/sem_open.c b/src/functional/sem_open.c
index 360e491..e2eeda3 100644
--- a/src/functional/sem_open.c
+++ b/src/functional/sem_open.c
@@ -7,10 +7,10 @@
 #include <time.h>
 #include <fcntl.h>
 #include <unistd.h>
-#include "test.h"
+#include <uk/test.h>//#include "test.h"
 
-#define TEST(c, ...) \
-	( (c) || (t_error(#c " failed: " __VA_ARGS__),0) )
+#define TEST(c, ...) UK_TEST_ASSERT((c))//\
+//	( (c) || (t_error(#c " failed: " __VA_ARGS__),0) )
 
 int main(void)
 {
@@ -52,5 +52,5 @@ int main(void)
 	TEST(sem_close(sem) == 0, "%s\n", strerror(errno));
 	TEST(sem_close(sem) == 0, "%s\n", strerror(errno));
 	TEST(sem_unlink(buf) == 0, "%s\n", strerror(errno));
-	return t_status;
+//	return t_status;
 }
-- 
2.25.1

