From 8a2fe63f6c5edad48f259e0e322e42f298cdeec8 Mon Sep 17 00:00:00 2001
From: Florin Postolache <florin.postolache80@gmail.com>
Date: Tue, 11 Oct 2022 09:18:33 +0300
Subject: [PATCH] Modify socket source to work with uktest

---
 src/functional/socket.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/functional/socket.c b/src/functional/socket.c
index b62cf98..98af443 100644
--- a/src/functional/socket.c
+++ b/src/functional/socket.c
@@ -6,12 +6,12 @@
 #include <netinet/in.h>
 #include <sys/time.h>
 #include <fcntl.h>
-#include "test.h"
+#include <uk/test.h>//"test.h"
 
-#define TEST(c, ...) ((c) ? 1 : (t_error(#c" failed: " __VA_ARGS__),0))
+#define TEST(c, ...) UK_TEST_ASSERT((c))//((c) ? 1 : (t_error(#c" failed: " __VA_ARGS__),0))
 #define TESTE(c) (errno=0, TEST(c, "errno = %s\n", strerror(errno)))
 
-int main(void)
+UK_TESTCASE(uk_libc_testsuite, socket_tests)//int main(void)
 {
 	struct sockaddr_in sa = { .sin_family = AF_INET };
 	int s, c, t;
@@ -54,5 +54,5 @@ int main(void)
 	close(c);
 	close(s);
 
-	return t_status;
+//	return t_status;
 }
-- 
2.25.1

