From 0d892e5f19c73d11beb408151060df52474c1231 Mon Sep 17 00:00:00 2001
From: Florin Postolache <florin.postolache80@gmail.com>
Date: Tue, 11 Oct 2022 22:36:15 +0300
Subject: [PATCH] Modify string_memmem source to work with uktest

---
 src/functional/string_memmem.c | 25 +++++++++++++++----------
 1 file changed, 15 insertions(+), 10 deletions(-)

diff --git a/src/functional/string_memmem.c b/src/functional/string_memmem.c
index 608eadf..3fc819a 100644
--- a/src/functional/string_memmem.c
+++ b/src/functional/string_memmem.c
@@ -2,26 +2,31 @@
 #define _GNU_SOURCE 1
 #endif
 #include <string.h>
-#include "test.h"
+#include <uk/test.h>//"test.h"
 
 #define N(s, tail, sub) { \
 	char *p = s tail; \
 	char *q = memmem(p, strlen(s), sub, strlen(sub)); \
-	if (q) \
-		t_error("memmem("#s" "#tail", %d, "#sub", %d) returned str+%d, wanted 0\n",\
-			strlen(s), strlen(sub), q-p); \
+	UK_TEST_EXPECT_NULL(q); \
 }
+//	if (q) \
+//		t_error("memmem("#s" "#tail", %d, "#sub", %d) returned str+%d, wanted 0\n",\
+//			strlen(s), strlen(sub), q-p); \
+//}
 
 #define T(s, sub, n) { \
 	char *p = s; \
 	char *q = memmem(p, strlen(p), sub, strlen(sub)); \
-	if (q == 0) \
-		t_error("memmem(%s,%s) returned 0, wanted str+%d\n", #s, #sub, n); \
-	else if (q - p != n) \
-		t_error("memmem(%s,%s) returned str+%d, wanted str+%d\n", #s, #sub, q-p, n); \
+	UK_TEST_EXPECT_NOT_ZERO(q); \
+	UK_TEST_ASSERT(q - p == n); \
 }
+//	if (q == 0) \
+//		t_error("memmem(%s,%s) returned 0, wanted str+%d\n", #s, #sub, n); \
+//	else if (q - p != n) \
+//		t_error("memmem(%s,%s) returned str+%d, wanted str+%d\n", #s, #sub, q-p, n); \
+//}
 
-int main(void)
+UK_TESTCASE(uk_libc_testsuite, string_memmem_tests)//int main(void)
 {
 	N("","a", "a")
 	N("a","a", "aa")
@@ -60,5 +65,5 @@ int main(void)
 	T("nanabanabanana", "banana", 8)
 	T("_ _\xff_ _", "_\xff_", 2)
 
-	return t_status;
+//	return t_status;
 }
-- 
2.25.1

