From ad1fc5424c969416eced347951f8e8129cc83b92 Mon Sep 17 00:00:00 2001
From: Florin Postolache <florin.postolache80@gmail.com>
Date: Thu, 10 Nov 2022 08:46:51 +0200
Subject: [PATCH] Modify mbsrtowcs-overflow source to work with uktest

Signed-off-by: Florin Postolache <florin.postolache80@gmail.com>
---
 src/regression/mbsrtowcs-overflow.c | 17 +++++------------
 1 file changed, 5 insertions(+), 12 deletions(-)

diff --git a/src/regression/mbsrtowcs-overflow.c b/src/regression/mbsrtowcs-overflow.c
index 6f2a9da..e51417b 100644
--- a/src/regression/mbsrtowcs-overflow.c
+++ b/src/regression/mbsrtowcs-overflow.c
@@ -1,22 +1,15 @@
 // commit 211264e46a2f1bc382a84435e904d1548de672b0 2013-09-27
 // mbsrtowcs should not write outside the ws array
 #include <wchar.h>
-#include "test.h"
+#include <uk/test.h>
 
-int main(void)
+UK_TESTCASE(uk_libc_string_tests, mbsrtowcs_overflow_tests)
 {
 	wchar_t ws[] = L"XXXXX";
 	const char *src = "abcd";
 	const char *want = src + 4;
-	size_t r;
 
-	r = mbsrtowcs(ws, &src, 4, 0);
-	if (r != 4)
-		t_error("mbsrtowcs(ws, &abcd, 4, 0) returned %zu, wanted 4\n", r);
-	if (src != want)
-		t_error("mbsrtowcs(ws, &abcd, 4, 0) set abcd to %p wanted %p\n", src, want);
-	if (wcscmp(ws, L"abcdX") != 0)
-		t_error("ws is L\"%ls\", wanted L\"abcdX\"\n", ws);
-
-	return t_status;
+	UK_TEST_ASSERT(mbsrtowcs(ws, &src, 4, 0) == 4);
+	UK_TEST_ASSERTF((src == want), "%p == %p", src, want);
+	UK_TEST_ASSERTF((wcscmp(ws, L"abcdX") == 0), "wcscmp(abcdX, abcdX) returned 0.");
 }
-- 
2.25.1

