From 366ec80bb384e99511f4f0ae42da5c30ec3395ba Mon Sep 17 00:00:00 2001
From: Florin Postolache <florin.postolache80@gmail.com>
Date: Thu, 10 Nov 2022 09:21:22 +0200
Subject: [PATCH] Modify memmem-oob source to work with uktest

Signed-off-by: Florin Postolache <florin.postolache80@gmail.com>
---
 src/regression/memmem-oob.c | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/src/regression/memmem-oob.c b/src/regression/memmem-oob.c
index 5249284..59293a3 100644
--- a/src/regression/memmem-oob.c
+++ b/src/regression/memmem-oob.c
@@ -4,12 +4,10 @@
 #define _GNU_SOURCE
 #endif
 #include <string.h>
-#include "test.h"
+#include <uk/test.h>
 
-int main(void)
+UK_TESTCASE(uk_libc_string_tests, memmem_oob_tests)
 {
 	char *p = memmem("abcde", 4, "cde", 3);
-	if (p)
-		t_error("memmem(abcde,4,cde,3) returned %s, want NULL\n", p);
-	return t_status;
+	UK_TEST_ASSERTF((p == NULL), "memmem(abcde, 4, cde, 3)");
 }
-- 
2.25.1

