From 95273315f0c16a05d86baaaef74e0a71bb835146 Mon Sep 17 00:00:00 2001
From: Florin Postolache <florin.postolache80@gmail.com>
Date: Sun, 23 Oct 2022 20:49:29 +0300
Subject: [PATCH] Modify popen source to work with uktest

Signed-off-by: Florin Postolache <florin.postolache80@gmail.com>
---
 src/functional/popen.c | 19 +++++++------------
 1 file changed, 7 insertions(+), 12 deletions(-)

diff --git a/src/functional/popen.c b/src/functional/popen.c
index 42c32c4..073398d 100644
--- a/src/functional/popen.c
+++ b/src/functional/popen.c
@@ -3,18 +3,14 @@
 #include <errno.h>
 #include <string.h>
 #include <signal.h>
-#include "test.h"
+#include <uk/test.h>
 
-#define TEST(r, f, x, m) ( \
-	((r) = (f)) == (x) || (t_error("%s failed (" m ")\n", #f, r, x), 0) )
+#define TEST(r, f, x, m) UK_TEST_ASSERT(((r) = (f)) == (x))
 
-#define TEST_E(f) ( \
-	(errno = 0), \
-	(f) || (t_error("%s failed (errno = %d)\n", #f, errno), 0) )
+#define TEST_E(f) errno = 0; \
+	UK_TEST_EXPECT_SNUM_GE((f), 1)
 
-#define TEST_S(s, x, m) ( \
-	!strcmp((s),(x)) || \
-		(t_error("[%s] != [%s] (%s)\n", s, x, m), 0) )
+#define TEST_S(s, x, m) UK_TEST_EXPECT_ZERO(strcmp((s),(x)))
 
 static sig_atomic_t got_sig;
 
@@ -22,7 +18,7 @@ static void handler(int sig) {
 	got_sig = 1;
 }
 
-int main(void)
+UK_TESTCASE(uk_libc_testsuite, popen_tests)
 {
 	int i;
 	char foo[6];
@@ -40,10 +36,9 @@ int main(void)
 	snprintf(cmd, sizeof cmd, "read a ; test \"x$a\" = xhello && kill -USR1 %d", getpid());
 	TEST_E(f = popen(cmd, "w"));
 	if (f) {
-		TEST_E(fputs("hello", f) >= 0);
+		TEST_E(fputs("hello", f));
 		TEST(i, pclose(f), 0, "exit status %04x != %04x");
 		TEST(i, got_sig, 1, "child process did not send signal");
 	}
 	signal(SIGUSR1, SIG_DFL);
-	return t_status;
 }
-- 
2.25.1

