From d117a944bc814eba6db19b0dc1158891950c0627 Mon Sep 17 00:00:00 2001
From: Florin Postolache <florin.postolache80@gmail.com>
Date: Sun, 23 Oct 2022 21:25:22 +0300
Subject: [PATCH] Modify pthread_cond source to work with uktest

Signed-off-by: Florin Postolache <florin.postolache80@gmail.com>
---
 src/functional/pthread_cond.c | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/src/functional/pthread_cond.c b/src/functional/pthread_cond.c
index a9a086a..fc09f7c 100644
--- a/src/functional/pthread_cond.c
+++ b/src/functional/pthread_cond.c
@@ -1,9 +1,8 @@
 #include <pthread.h>
 #include <string.h>
-#include "test.h"
+#include <uk/test.h>
 
-#define TEST(r, f, m) ( \
-	((r) = (f)) == 0 || (t_error("%s failed: %s (" m ")\n", #f, strerror(r)), 0) )
+#define TEST(r, f, m) UK_TEST_EXPECT_ZERO((f))
 
 
 static void *start_signal(void *arg)
@@ -29,7 +28,7 @@ static void *start_wait(void *arg)
 	return 0;
 }
 
-int main(void)
+UK_TESTCASE(uk_libc_pthread_tests, pthread_cond_test)
 {
 	pthread_t td, td1, td2, td3;
 	int r;
@@ -95,5 +94,4 @@ int main(void)
 	TEST(r, pthread_mutex_destroy(&mtx), "");
 	TEST(r, pthread_cond_destroy(&cond), "");
 
-	return t_status;
 }
-- 
2.25.1

