From 1e6ce18610f8e80c375ca1332c8f9c6a81096625 Mon Sep 17 00:00:00 2001
From: Florin Postolache <florin.postolache80@gmail.com>
Date: Sun, 23 Oct 2022 22:18:17 +0300
Subject: [PATCH] Modify pthread_tsd source to work with uktest

Signed-off-by: Florin Postolache <florin.postolache80@gmail.com>
---
 src/functional/pthread_tsd.c | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/src/functional/pthread_tsd.c b/src/functional/pthread_tsd.c
index 6e6360d..718e181 100644
--- a/src/functional/pthread_tsd.c
+++ b/src/functional/pthread_tsd.c
@@ -1,10 +1,9 @@
 #include <pthread.h>
 #include <string.h>
-#include "test.h"
+#include <uk/test.h>
 
-#define TESTC(c, m) ( (c) || (t_error("%s failed (" m ")\n", #c), 0) )
-#define TESTR(r, f, m) ( \
-	((r) = (f)) == 0 || (t_error("%s failed: %s (" m ")\n", #f, strerror(r)), 0) )
+#define TESTC(c, m) UK_TEST_ASSERT((c))
+#define TESTR(r, f, m) UK_TEST_EXPECT_ZERO(((r) = (f)))
 
 static pthread_key_t k1, k2;
 
@@ -21,7 +20,7 @@ static void *start(void *arg)
 	return 0;
 }
 
-int main(void)
+UK_TESTCASE(uk_libc_pthread_tests, pthread_tsd_test)
 {
 	pthread_t td;
 	int r;
@@ -45,5 +44,4 @@ int main(void)
 	TESTR(r, pthread_setspecific(k2, 0), "failed to clear tsd");
 	TESTR(r, pthread_key_delete(k1), "failed to destroy key");
 	TESTR(r, pthread_key_delete(k2), "failed to destroy key");
-	return t_status;
 }
-- 
2.25.1

