From d05c0cdaddaa61f6898114ca62639e361c1b473b Mon Sep 17 00:00:00 2001
From: Florin Postolache <florin.postolache80@gmail.com>
Date: Fri, 28 Oct 2022 12:47:59 +0300
Subject: [PATCH] Modify regex-bracket-icase source to work with uktest

Signed-off-by: Florin Postolache <florin.postolache80@gmail.com>
---
 src/regression/regex-bracket-icase.c | 11 ++++-------
 1 file changed, 4 insertions(+), 7 deletions(-)

diff --git a/src/regression/regex-bracket-icase.c b/src/regression/regex-bracket-icase.c
index 82b9287..41b3261 100644
--- a/src/regression/regex-bracket-icase.c
+++ b/src/regression/regex-bracket-icase.c
@@ -3,9 +3,9 @@
 #include <regex.h>
 #include <limits.h>
 #include <stdio.h>
-#include "test.h"
+#include <uk/test.h>
 
-int main(void)
+UK_TESTCASE(uk_libc_testsuite, regex_bracket_icase_tests)
 {
 	char buf[100];
 	char *pat;
@@ -28,19 +28,16 @@ int main(void)
 
 	pat = "[^aBcC]";
 	n = regcomp(&re, pat, REG_ICASE);
+	UK_TEST_EXPECT_ZERO(n);
 	if (n) {
 		regerror(n, &re, buf, sizeof buf);
-		t_error("regcomp(\"%s\") failed: %d (%s)\n", pat, n, buf);
 	}
 
 	for (i = 0; t[i].s; i++) {
 		n = regexec(&re, t[i].s, 0, 0, 0);
+		UK_TEST_ASSERT(n == t[i].n);
 		if (n != t[i].n) {
 			regerror(n, &re, buf, sizeof buf);
-			t_error("regexec(/%s/, \"%s\") returned %d (%s), wanted %d\n",
-				pat, t[i].s, n, buf, t[i].n);
 		}
 	}
-
-	return t_status;
 }
-- 
2.25.1

