From 483d3579fb8e8983fbf82d49d8013c1b799d982a Mon Sep 17 00:00:00 2001
From: Florin Postolache <florin.postolache80@gmail.com>
Date: Wed, 16 Nov 2022 15:00:32 +0200
Subject: [PATCH] Modify search_insque source to work with uktest

Signed-off-by: Florin Postolache <florin.postolache80@gmail.com>
---
 src/functional/search_insque.c | 14 +++++---------
 1 file changed, 5 insertions(+), 9 deletions(-)

diff --git a/src/functional/search_insque.c b/src/functional/search_insque.c
index 0f4c370..3e36939 100644
--- a/src/functional/search_insque.c
+++ b/src/functional/search_insque.c
@@ -3,7 +3,7 @@
 #endif
 #include <stdlib.h>
 #include <search.h>
-#include "test.h"
+#include <uk/test.h>
 
 struct q {
 	struct q *n;
@@ -18,7 +18,7 @@ static struct q *new(int i)
 	return q;
 }
 
-int main()
+UK_TESTCASE(uk_libc_structure_search_tests, search_insque_tests)
 {
 	struct q *q = new(0);
 	struct q *p;
@@ -31,14 +31,10 @@ int main()
 	}
 	p = q;
 	while (q) {
-		if (q->i != --i)
-			t_error("walking queue: got %d, wanted %d\n", q->i, i);
+		UK_TEST_ASSERTF(q->i == --i, "The element at position %d in queue is %d.", q->i, q->i);
 		q = q->p;
 	}
 	remque(p->p);
-	if (p->p->i != p->i-2)
-		t_error("remque: got %d, wanted %d\n", p->p->i, p->i-2);
-	if (p->p->n->i != p->i)
-		t_error("remque: got %d, wanted %d\n", p->p->n->i, p->i);
-	return t_status;
+	UK_TEST_ASSERTF(p->p->i == p->i-2, "After removing element 8, the previous element of %d is %d.", p->i, p->p->i);
+	UK_TEST_ASSERT(p->p->n->i == p->i);
 }
-- 
2.25.1

