From b47d95993be8cadbbb4389ad1c2785819fc806be Mon Sep 17 00:00:00 2001
From: Florin Postolache <florin.postolache80@gmail.com>
Date: Mon, 24 Oct 2022 22:44:06 +0300
Subject: [PATCH] Modify setjmp source to work with uktest

Signed-off-by: Florin Postolache <florin.postolache80@gmail.com>
---
 src/functional/setjmp.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/src/functional/setjmp.c b/src/functional/setjmp.c
index 7ba09d3..67a34aa 100644
--- a/src/functional/setjmp.c
+++ b/src/functional/setjmp.c
@@ -1,11 +1,11 @@
 #include <string.h>
 #include <signal.h>
 #include <setjmp.h>
-#include "test.h"
+#include <uk/test.h>
 
-#define TEST(c, ...) ((c) ? 1 : (t_error(#c" failed: " __VA_ARGS__),0))
+#define TEST(c, ...) UK_TEST_ASSERT((c))
 
-int main(void)
+UK_TESTCASE(uk_libc_setjmp_tests, setjmp_tests)
 {
 	volatile int x = 0, r;
 	jmp_buf jb;
@@ -61,5 +61,4 @@ int main(void)
 	sigprocmask(SIG_SETMASK, &set, &set2);
 	TEST(sigismember(&set2, SIGUSR1)==1, "siglongjmp incorrectly restored mask\n");
 
-	return t_status;
 }
-- 
2.25.1

