From 3e66a977115156739a72fd82e999ec2f3ee8adb3 Mon Sep 17 00:00:00 2001
From: Florin Postolache <florin.postolache80@gmail.com>
Date: Tue, 8 Nov 2022 22:35:26 +0200
Subject: [PATCH] Modify string_strcspn source to work with uktest

Signed-off-by: Florin Postolache <florin.postolache80@gmail.com>
---
 src/functional/string_strcspn.c | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/src/functional/string_strcspn.c b/src/functional/string_strcspn.c
index 88df383..87ffe01 100644
--- a/src/functional/string_strcspn.c
+++ b/src/functional/string_strcspn.c
@@ -1,16 +1,15 @@
 #include <stddef.h>
 #include <string.h>
-#include "test.h"
+#include <uk/test.h>
 
 #define T(s, c, n) { \
 	char *p = s; \
 	char *q = c; \
 	size_t r = strcspn(p, q); \
-	if (r != n) \
-		t_error("strcspn(%s,%s) returned %lu, wanted %lu\n", #s, #c, (unsigned long)r, (unsigned long)(n)); \
+	UK_TEST_ASSERTF((r ==n), "Strcspn found %d characters that are not in q.", n); \
 }
 
-int main(void)
+UK_TESTCASE(uk_libc_string_tests, string_strcspn_tests)
 {
 	int i;
 	char a[128];
@@ -29,6 +28,4 @@ int main(void)
 	T("abc", a, 0)
 	T("\xff\x80 abc", a, 2)
 	T(s, "\xff", 254)
-
-	return t_status;
 }
-- 
2.25.1

