From 48da3045a78106d1c838e4c3aebf11b507a802f1 Mon Sep 17 00:00:00 2001
From: Florin Postolache <florin.postolache80@gmail.com>
Date: Tue, 8 Nov 2022 22:56:03 +0200
Subject: [PATCH] Modify string_strstr source to work with uktest

Signed-off-by: Florin Postolache <florin.postolache80@gmail.com>
---
 src/functional/string_strstr.c | 15 +++++----------
 1 file changed, 5 insertions(+), 10 deletions(-)

diff --git a/src/functional/string_strstr.c b/src/functional/string_strstr.c
index 54e0fd5..4eee866 100644
--- a/src/functional/string_strstr.c
+++ b/src/functional/string_strstr.c
@@ -1,23 +1,20 @@
 #include <string.h>
-#include "test.h"
+#include <uk/test.h>
 
 #define N(s, sub) { \
 	char *p = s; \
 	char *q = strstr(p, sub); \
-	if (q) \
-		t_error("strstr(%s,%s) returned str+%d, wanted 0\n", #s, #sub, q-p); \
+	UK_TEST_ASSERTF((q==NULL),"Strstr returned NULL."); \
 }
 
 #define T(s, sub, n) { \
 	char *p = s; \
 	char *q = strstr(p, sub); \
-	if (q == 0) \
-		t_error("strstr(%s,%s) returned 0, wanted str+%d\n", #s, #sub, n); \
-	else if (q - p != n) \
-		t_error("strstr(%s,%s) returned str+%d, wanted str+%d\n", #s, #sub, q-p, n); \
+	UK_TEST_ASSERTF((q != NULL), "Strstr didn't return NULL."); \
+	UK_TEST_ASSERTF((q - p == n), "Strstr returned the correct position."); \
 }
 
-int main(void)
+UK_TESTCASE(uk_libc_string_tests, string_strstr_tests)
 {
 	N("", "a")
 	N("a", "aa")
@@ -53,6 +50,4 @@ int main(void)
 	T("nanabanabanana", "anab", 1)
 	T("nanabanabanana", "banana", 8)
 	T("_ _\xff_ _", "_\xff_", 2)
-
-	return t_status;
 }
-- 
2.25.1

