From bd2f6e9d7d1b222ba416379165b71e5be96f4e52 Mon Sep 17 00:00:00 2001
From: Florin Postolache <florin.postolache80@gmail.com>
Date: Tue, 25 Oct 2022 10:10:29 +0300
Subject: [PATCH] Modify string_strstr source to work with uktest

Signed-off-by: Florin Postolache <florin.postolache80@gmail.com>
---
 src/functional/string_strstr.c | 15 +++++----------
 1 file changed, 5 insertions(+), 10 deletions(-)

diff --git a/src/functional/string_strstr.c b/src/functional/string_strstr.c
index 54e0fd5..70be923 100644
--- a/src/functional/string_strstr.c
+++ b/src/functional/string_strstr.c
@@ -1,23 +1,20 @@
 #include <string.h>
-#include "test.h"
+#include <uk/test.h>
 
 #define N(s, sub) { \
 	char *p = s; \
 	char *q = strstr(p, sub); \
-	if (q) \
-		t_error("strstr(%s,%s) returned str+%d, wanted 0\n", #s, #sub, q-p); \
+	UK_TEST_EXPECT_NULL(q); \
 }
 
 #define T(s, sub, n) { \
 	char *p = s; \
 	char *q = strstr(p, sub); \
-	if (q == 0) \
-		t_error("strstr(%s,%s) returned 0, wanted str+%d\n", #s, #sub, n); \
-	else if (q - p != n) \
-		t_error("strstr(%s,%s) returned str+%d, wanted str+%d\n", #s, #sub, q-p, n); \
+	UK_TEST_EXPECT_NOT_NULL(q); \
+	UK_TEST_ASSERT((q - p == n)); \
 }
 
-int main(void)
+UK_TESTCASE(uk_libc_testsuite, string_strstr_tests)
 {
 	N("", "a")
 	N("a", "aa")
@@ -53,6 +50,4 @@ int main(void)
 	T("nanabanabanana", "anab", 1)
 	T("nanabanabanana", "banana", 8)
 	T("_ _\xff_ _", "_\xff_", 2)
-
-	return t_status;
 }
-- 
2.25.1

