From d9d99ce7c13e61dd0f2255901d2d67c3f4c44cca Mon Sep 17 00:00:00 2001
From: Florin Postolache <florin.postolache80@gmail.com>
Date: Tue, 25 Oct 2022 12:31:08 +0300
Subject: [PATCH] Modify tls_init source to work with uktest

Signed-off-by: Florin Postolache <florin.postolache80@gmail.com>
---
 src/functional/tls_init.c | 26 ++++++++++++--------------
 1 file changed, 12 insertions(+), 14 deletions(-)

diff --git a/src/functional/tls_init.c b/src/functional/tls_init.c
index d8b6e8f..579febf 100644
--- a/src/functional/tls_init.c
+++ b/src/functional/tls_init.c
@@ -1,5 +1,5 @@
 #include <pthread.h>
-#include "test.h"
+#include <uk/test.h>
 
 __thread int tls_fix = 23;
 __thread int tls_zero;
@@ -7,26 +7,24 @@ __thread int tls_zero;
 static void *f(void *arg)
 {
 	if (tls_fix != 23)
-		t_error("fixed init failed: want 23 got %d\n", tls_fix);
+		return -1;
 	if (tls_zero != 0)
-		t_error("zero init failed: want 0 got %d\n", tls_zero);
+		return -1;
 	tls_fix++;
 	tls_zero++;
 	return 0;
 }
 
-#define CHECK(f) do{ if(f) t_error("%s failed.\n", #f); }while(0)
+#define CHECK(f) do{ UK_TEST_EXPECT_ZERO(f); }while(0)
 #define length(a) (sizeof(a)/sizeof*(a))
 
-int main()
+UK_TESTCASE(uk_libc_pthread_tests, tls_init_tests)
 {
 	pthread_t t[5];
-	int i, j;
+	int i, j, res[5];
 
-	if (tls_fix != 23)
-		t_error("fixed init failed: want 23 got %d\n", tls_fix);
-	if (tls_zero != 0)
-		t_error("zero init failed: want 0 got %d\n", tls_zero);
+	UK_TEST_ASSERT(tls_fix == 23);
+	UK_TEST_ASSERT(tls_zero == 0);
 
 	for (j = 0; j < 2; j++) {
 		for (i = 0; i < length(t); i++) {
@@ -34,9 +32,9 @@ int main()
 			tls_fix++;
 			tls_zero++;
 		}
-		for (i = 0; i < length(t); i++)
-			CHECK(pthread_join(t[i], 0));
+		for (i = 0; i < length(t); i++){
+			CHECK(pthread_join(t[i], (void *)&res[i]));
+			UK_TEST_EXPECT_ZERO(res[i]);
+		}
 	}
-
-	return t_status;
 }
-- 
2.25.1

